cmake/wannier90.pc.in

```diff
@@ -0,0 +1,9 @@
+prefix: @CMAKE_INSTALL_PREFIX@
+
+Name: wannier
+Version: @PROJECT_VERSION@
+Description: @PROJECT_DESCRIPTION@
+URL: @PROJECT_HOMEPAGE_URL@
+
+Cflags: -I@CMAKE_INSTALL_FULL_MODULEDIR@ -I@CMAKE_INSTALL_FULL_INCLUDEDIR@
+Libs: -L@CMAKE_INSTALL_FULL_LIBDIR@ -lwannier90
```

Comment:
```suggestion
prefix: @CMAKE_INSTALL_PREFIX@
libdir: ${prefix}/@CMAKE_INSTALL_LIBDIR@
includedir: ${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
moduledirL: ${prefix}/@WANNIER90_INSTALL_MODULEDIR@

Name: wannier
Version: @PROJECT_VERSION@
Description: @PROJECT_DESCRIPTION@
URL: @PROJECT_HOMEPAGE_URL@

Cflags: -I${moduledir} -I${includedir}
Libs: -L${libdir} -lwannier90
```
Using full paths make it unrelocatable.

Reply:
Indeed, there was some context for this, but I completely lost it. Maybe it was related to testing the pkg-config file in the integration tests since the file in the build directory is not usable. Really don't remember why I made it like that.

There are more issues there though because `PUBLIC` linked libraries are not included. In principle this should be written as:
```
file(GENERATE OUTPUT wannier.pc.in INPUT wannier.pc.in.in)
configure_file(${CMAKE_CURRENT_BINARY_DIR}/wannier.pc.in wannier.pc)
```
And use a gen-ex depending on the target property to get the `-l`/`-L`/`-I` flags needed there. Didn't get the time to investigate what are appropriate flags for it though, would appreciate some help/discussion on this.

This is necessary because of the `use mpi_f08` that is conditionally used in the Fortran module api.

Reply:
You could add a new string variable `WANNIER_LINK_LIBRARIES` and add any library wannier is linked to. then add `@WANNIER_LINK_LIBRARIES@` to `wannier.pc.in` file.

Reply:
The question is where do we get the values to construct it? If we figure out that, I can write a prototype to be more templatable.

These values are taken from targets like `MPI::MPI_Fortran`. We can investigate that is populate in there with things like: https://stackoverflow.com/a/34292622. It should be in an `*_INTERFACE_*` like property